tools/bin/commands/util/operator_funcs

```diff
@@ -113,58 +113,25 @@ build_operator_local() {
 
 build_operator_docker() {
     local operator_dir="${1}"
-    local as_user="${2:-}"
-    local do_skip_tests="${3:-}"
+    local do_skip_tests="${2:-}"
 
-    echo "Creating syndesis-operator with $GO_BUILDER_IMAGE"
-    local extra_opts=""
-    if [ -n "$as_user" ]; then
-        extra_opts="$extra_opts -u $as_user -e XDG_CACHE_HOME=.cache"
-    else
-        extra_opts="$extra_opts -u $(id -u):$(id -g)"
-    fi
-
-    local cmd="(echo Compiling syndesis-operator; go build ./cmd/syndesis-operator)"
-
-    if [ ! "$do_skip_tests" ]; then
-        cmd="(echo Running tests; go test $TEST_TARGETS) && $cmd"
-    fi
-
-    if [ $(need_to_ensure_dep) == 1 ]; then
-        cmd="(echo Running 'dep ensure'; dep ensure -vendor-only -v) && $cmd"
-    fi
-
-    local docker_gopath_volume=$(mktemp -d ${TMPDIR:-/tmp}/gopath.XXXXX)
-    local docker_cache_volume=$(mktemp -d ${TMPDIR:-/tmp}/gocache.XXXXX)
-    trap "rm -rf $docker_cache_volume $docker_gopath_volume" EXIT
-
-    mkdir_and_check_owner "$operator_dir/dep-cache"
-    mkdir_and_check_owner "$docker_cache_volume"
-    mkdir_and_check_owner "$docker_gopath_volume/pkg/dep"
-    mkdir_and_check_owner "$docker_gopath_volume/src/github.com/syndesisio/syndesis/install/operator"
-
-    docker run -w /gopath/src/github.com/syndesisio/syndesis/install/operator \
-               $extra_opts \
-               -v "$docker_gopath_volume":/gopath:z \
-               -v "$docker_cache_volume":/.cache:z \
-               -v $operator_dir/dep-cache:/gopath/pkg/dep:z \
-               -v $operator_dir:/gopath/src/github.com/syndesisio/syndesis/install/operator:z \
-               -e CGO_ENABLED=0 \
-               $GO_BUILDER_IMAGE \
-               sh -c "$cmd"
+    cd $operator_dir
+    docker build -t syndesis-operator-builder . -f Dockerfile-builder
+    docker run syndesis-operator-builder cat /syndesis-operator > syndesis-operator
```

Comment:
Nice trick, but I would prefer 'docker cp' on the finished builder container to avoid potential encoding / pipe issues on different Linux. I think it makes it cleaner and a bit more robust to use native Docker means:

```bash
local id=$(docker create syndesis-operator-builder)
docker cp $id:/syndesis-operator syndesis-operator
```

Reply:
Why would encoding be taking place?  I've not seen any issues with encoding BTW.  Also we would need to delete the container right?  Doing a quick run seems easier.

Reply:
`docker run` is `docker create` + `docker start` and you don't clean up the container either afterwards (you could add a --rm though).

As mentioned, it's just a gut feeling to stick to a `docker cp` instead of a `cat` as feels more binary-ish, but if you insist of doing it that way, I'm fine, too. Its an easy change we could add if it breaks anyway, no need to waste energy here.